app/controllers/api/base_controller/parameters.rb

```diff
@@ -175,6 +177,18 @@ def sort_directive(attr, order, options)
         sort_item << " DESC" if order && order.downcase.start_with?("desc")
         sort_item
       end
+
+      def sort_virtual_attr_directive(klass, attr, order, options)
+        raise BadRequestError,
+              "#{klass.name} cannot be sorted by #{attr}" unless klass.attribute_supported_by_sql?(attr)
+        raise BadRequestError,
+              "#{klass.name} cannot be sorted with ignored case" if options.map(&:downcase).include?("ignore_case")
+        if order
+          { attr.to_sym => order.downcase }
```

Comment:
maybe something like:

```ruby
klass.arel_attribute(attr).downcase.send(order.downcase)
```

I wonder if there is a way for us to reuse [miq_report/search.rb#get_order_info](https://github.com/ManageIq/manageiq/blob/master/app/models/miq_report/search.rb#L43)

Also, a while ago, `arel_attribute().lower` was not supported for either attributes or virtual attributes - can't quite remember. So instead of the `Arel::Nodes::Lower` you may want to just try `.lower` and see if it will work